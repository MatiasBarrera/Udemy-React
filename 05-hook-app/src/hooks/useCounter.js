import { useState } from 'react'

export const useCounter = (inicialState = 10) => {
  const [counter, setCounter] = useState(inicialState)

  const increment = () => setCounter(counter + 1)
  const decrement = () => {
    const isLessThanZero = counter - 1 === 0
    isLessThanZero ? setCounter(1) : setCounter(counter - 1)
  }
  const reset = () => setCounter(inicialState)

  return { counter, increment, decrement, reset }
}
