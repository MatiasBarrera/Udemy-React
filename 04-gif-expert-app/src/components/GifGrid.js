import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifts(category)
  return (
    <>
      <h3 className="animate__animated animate__bounceIn">{category}</h3>

      {loading && <p className="animate__animated flash">Loading</p>}

      <div className="card-grid">
        {images.map(img => <GifGridItem key={img.id} {...img} />)}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid
