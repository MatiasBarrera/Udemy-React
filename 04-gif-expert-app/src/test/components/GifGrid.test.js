import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifts } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Prueba de componente GifGrid.js', () => {
  const category = 'hunter x hunter'
  //let wrapper = shallow(<GifGrid category={category}/>)

  test('Debe de mostrar el componente <GifGrid /> correctamente', () => {
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });
    
    const wrapper = shallow(<GifGrid category={category}/>)
    expect(wrapper).toMatchSnapshot()
  });

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://google.com',
      title: 'ABC Title'
    }]
  
    useFetchGifts.mockReturnValue({
      data: gifs,
      loading: false,
    });
  
    const wrapper = shallow(<GifGrid category={category}/>);

    const p = wrapper.find('p').exists()
    const gifGridItem = wrapper.find('GifGridItem').length

    expect(p).toBeFalsy()
    expect(gifGridItem).toBe(gifs.length)

  });
});