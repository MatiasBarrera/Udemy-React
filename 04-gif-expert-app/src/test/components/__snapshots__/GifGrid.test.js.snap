// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifGrid.js Debe de mostrar el componente <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceIn"
  >
    hunter x hunter
  </h3>
  <p
    className="animate__animated flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
